### ISSUE TITLE
Is the 84x84 a only input resolution for agent in submission?

### ISSUE THREAD

**Issue by Denys88** [beyretb/AnimalAI-Olympics#54]
When I learning I can choose any resolution I want.
Can I change it for inference?
In testDocker.py I see this code:
    env = AnimalAIEnv(
        environment_filename='/aaio/test/env/AnimalAI',
        seed=0,
        retro=False,
        n_arenas=1,
        worker_id=1,
        docker_training=True,
    )

Does it mean that only possible resolution is 84x84?

**Comment by MLanghof** [comment]
You will be tested on 84x84.

https://github.com/beyretb/AnimalAI-Olympics/issues/27#issuecomment-507362296

> We've added an option to change the resolution of the agent. It's still square and can vary between 8x8 and 512x512 (defaults to the Atari 84x84 as this is what you'll be tested on).

**Comment by beyretb** [comment]
Hello,

We have now added the possibility to specify the resolution of the environment during evaluation. You can now add an attribute `self.resolution` to your agent in `__init__`. This value has to be between 4 and 256 (the training environment goes to 512 but we had to limit to 256 for evaluation due to compute constraints). See the documentation [here](https://github.com/beyretb/AnimalAI-Olympics/blob/master/documentation/submission.md#python-agent-and-associated-data)

Remember that the bigger the observation, the longer it takes to run a single step, thus giving your agent less time for inference.

**Comment by Denys88** [comment]
Thanks,
I got better score even on 96x96 resolution during my local validation :)

**Comment by krisives** [comment]
@beyretb Just for clarification, can this resolution be changed between episodes or is it only set once when the agent is first loaded?

**Comment by beyretb** [comment]
@krisives yes only once when we load the environment

**Comment by dan9thsense** [comment]
This is a HUGE change to the competition.  Networks trained at the 84x84 resolution will have to be re-trained and that factor alone is a big time sink.  This is the sort of parameter that, announced at the start of the competition as a fixed value for final submission, should not be changed half-way through.

**Comment by krisives** [comment]
I agree that while it's nice to have a new resolution options available having them sprung on us midway through the competition was a bit of a pain because we have lots of models already trained at 84x84.

**Comment by jaggiK** [comment]
Changing the resolution in the middle of the competition does not seem to be reasonable when so much time and effort has been put working on 84x84. It is a pain going through the process all over again.

**Comment by mdcrosby** [comment]
We thought that allowing for more resolutions would only increase possibilities for people entering the competition and that 84x84 would still be a good resolution to use, therefore not changing anyone's plans. However, we overlooked that this would be unfair to anyone who had invested time in 84x84 that would now want to have explored the other options. We had not intended this change affects those that have already been engaging with the competition and are therefore going to remove this from the evaluation script. You can still use multiple resolutions for *training*, but *evaluation* will require 84x84.

The time increase will remain. This should have very minimal effect as very few entries come even close to the time limit. It also allows for a wider variety of approaches to be entered. Lastly, it potentially helps those that have access to less resources compared to the resolution change which primarily helped those with access to more resources.

Please be aware that in our rules the final evaluation (In November not for AWS) may include extra variations on the tests (resources permitting, we will run them more than once with minor variations to get a more robust score) and therefore the final score may not be identical with the score on the current leaderboard.

**Comment by Denys88** [comment]
Now I invested time into other resolution. 
What to do with leaderboard right now?
84 seems to be pretty low resolution. I was only able to play somehow starting from 128.
I lost two days of learning on new resolution.
And I see 'The resolution provided must be 84' error.
Please think about your decisions.

**Comment by mdcrosby** [comment]
It is regrettable that we made this mistake, but, given that we put ourselves in that situation, it became clear that we needed to respect that there has been 1.5 months of work from multiple teams that has been performed with the understanding that the final test will be performed at 84x84 resolution. If we could add the extra functionality without undoing the 1.5 months of work then we would, and we are thinking about some alternatives, but it is unlikely to be possible for this iteration of the competition.

It goes without saying that we will be much more careful in future before adding any extra functionality (and we do not intend to make any further changes unless we need to fix bugs).
